src/rdkafka_assignor.h

```diff
@@ -129,9 +142,9 @@ void rd_kafka_group_member_clear (rd_kafka_group_member_t *rkgm);
  * rd_kafka_range_assignor.c
  */
 rd_kafka_resp_err_t
-rd_kafka_range_assignor_assign_cb (rd_kafka_t *rk,
+rd_kafka_range_assignor_assign_cb (rd_kafka_assignor_t *rkas,
```

Comment:
rk is generally the first argument.

Can we make rkas const?

Reply:
`const`: yep. assignors are effectively a constant part of the configuration. there should be const in a lot more places (have made the changes).

`rk`: it generally makes sense for it to be the first argument because it's the focus of many methods (plays the role of a `self`). does this apply here? `rk` is used by the existing assignors only to enable logging of debug messages. Given the java assignor classes actually have no dependency on the client instance, that is good to see. I think it's a bit unfortunate it's a parameter at all, and possibly even misleading to make it the first parameter. that said, it looks weird to me too as is and i'll change it (but see prev comments). the ideal interface would have a `rd_logger_t` or something i think (and that would be one of the later parameters). that seems not-practical/ideal to add at this point. commentary?

Reply:
i put the `rkas` first because the method is most closely related to this. the relationship isn't as strong as if `rkas` managed dynamic state, but it's still pretty strong. anyway, the root cause of the difficult decision here is that `rk` doesn't really belong.